src/WebJobs.Extensions.DurableTask/ContextInterfaces/DurableContextExtensions.cs

```diff
@@ -405,11 +406,49 @@ public static Task CallEntityAsync(this IDurableOrchestrationContext context, En
         public static Task<string> StartNewAsync(
             this IDurableOrchestrationClient client,
             string orchestratorFunctionName,
-            object input)
+            string instanceId)
+        {
+            return client.StartNewAsync<object>(orchestratorFunctionName, instanceId, null);
+        }
+
+        /// <summary>
+        /// Starts a new execution of the specified orchestrator function.
+        /// </summary>
+        /// <param name="client">The client object.</param>
+        /// <param name="orchestratorFunctionName">The name of the orchestrator function to start.</param>
+        /// <param name="input">JSON-serializeable input value for the orchestrator function.</param>
+        /// <typeparam name="T">The type of the input value for the orchestrator function.</typeparam>
+        /// <returns>A task that completes when the orchestration is started. The task contains the instance id of the started
+        /// orchestratation instance.</returns>
+        /// <exception cref="ArgumentException">
+        /// The specified function does not exist, is disabled, or is not an orchestrator function.
+        /// </exception>
+        public static Task<string> StartNewAsync<T>(
+            this IDurableOrchestrationClient client,
+            string orchestratorFunctionName,
+            T input)
+            where T : class
         {
             return client.StartNewAsync(orchestratorFunctionName, string.Empty, input);
         }
 
+        /// <summary>
+        /// Starts a new execution of the specified orchestrator function.
+        /// </summary>
+        /// <param name="client">The client object.</param>
+        /// <param name="orchestratorFunctionName">The name of the orchestrator function to start.</param>
+        /// <returns>A task that completes when the orchestration is started. The task contains the instance id of the started
+        /// orchestratation instance.</returns>
+        /// <exception cref="ArgumentException">
+        /// The specified function does not exist, is disabled, or is not an orchestrator function.
+        /// </exception>
+        public static Task<string> StartNewAsync(
```

Comment:
nit: I would put this overload at the top of the list since it's the least specific.